engine/message/ratelimit.go

```diff
@@ -0,0 +1,52 @@
+package message
+
+import (
+	"time"
+
+	"github.com/target/goalert/notification"
+)
+
+// GlobalCMThrottle represents the rate limits for each notification type.
+var GlobalCMThrottle ThrottleConfig = ThrottleRules{{Count: 5, Per: 5 * time.Second}}
+
+// PerCMThrottle configures rate limits for individual contact methods.
+var PerCMThrottle ThrottleConfig
+
+func init() {
+	var perCM ThrottleConfigBuilder
+
+	// all message types
+	perCM.AddRules([]ThrottleRule{{Count: 1, Per: time.Minute}})
+
+	// status notifications
+	perCM.
+		WithMsgTypes(notification.MessageTypeAlertStatus, notification.MessageTypeAlertStatusBundle).
+		AddRules([]ThrottleRule{
+			{Count: 1, Per: 3 * time.Minute},
+			{Count: 3, Per: 20 * time.Minute},
+			{Count: 8, Per: 120 * time.Minute, Smooth: true},
+		})
+
+	// alert notifications
+	alertMessages := perCM.WithMsgTypes(notification.MessageTypeAlert, notification.MessageTypeAlertBundle)
+
+	alertMessages.
+		WithDestTypes(notification.DestTypeVoice).
+		AddRules([]ThrottleRule{
+			{Count: 1, Per: time.Minute},
+			{Count: 3, Per: 15 * time.Minute},
+			{Count: 7, Per: time.Hour, Smooth: true},
```

Comment:
Should we add a short comment here or upate the values in the PR description? If the values change because of `Smooth: true` might be helpful to have a one-liner.

Reply:
I'll update the PR description, the struct has a comment for what Smooth does already